Sources/MongoSwift/BSON/Document.swift

```diff
@@ -193,12 +193,30 @@ public struct Document: ExpressibleByDictionaryLiteral, ExpressibleByArrayLitera
      *  d["a"] = 1
      *  print(d["a"]) // prints 1
      *  ```
+     * A nil return suggests that the subscripted key does not exist in the `Document`. A true BSON null is returned as
+     * an `NSNull`.
      */
     public subscript(key: String) -> BSONValue? {
-        get { return DocumentIterator(forDocument: self, advancedTo: key)?.currentValue }
+        get {
+            let iterOptional = DocumentIterator(forDocument: self)
```

Comment:
nit: just unwrap the initializer call directly - 
```swift
guard let iter = DocumentIterator(forDocument: self) else { 
```

Reply:
Oh, thanks! Not sure why I didn't do this before.

Done.